pg_auth_mon.c

```diff
@@ -191,6 +194,7 @@ auth_monitor(Port *port, int status)
 		fai->key = key;
 		memset(&fai->total_successful_attempts, 0, sizeof(auth_mon_rec)
 			   - offsetof(auth_mon_rec, total_successful_attempts));
+		namestrcpy(&fai->initial_rolename, port->user_name);
```

Comment:
I think it is not enough to have the `initial_rolename`, because it will not cover role renames.
I.e.:
1. user1 logs in
2. user1 is renamed to user2
3. user2 logs in
4. user2 is dropped

After that the view will report "user1", despite the last successful or failed attempt was from the "user2".

Reply:
I know about the renames, my plan was to come up minimal working version is to extend it later